src/core/PartitionAware.ts

```diff
@@ -0,0 +1,22 @@
+/**
```

Comment:
I know this is a straight copy of the Javadoc, but I still think this is poorly worded. 

```PartitionAware means that data will be based in the same member based on the partition key
and implementing tasks will be executed on the {@link getPartitionKey}'s owner member.
```

I would phrase it like this:

Implementing PartitionAware allows one to override the default partitioning scheme. Instead of using the keys themselves to spread the data around the cluster the key returned by `getPartitionKey()` is used. 

This provides the user with an ability to contain related keys within the same partition and, consequently, within the same node. 


`Data and execution occurs on the same partition.`

What is "data occurring"?

I would omit any references to an executor service since it is not likely we will have it in Node client.